### DISCUSSION TITLE
How to know all parameters and attributes acceptable by Cloud Pricing API?

### DISCUSSION THREAD

**Discussion by vitor-antoni** [infracost/infracost#3171] [Q&A]
Hi there.

I have one doubt about how could I know all parameters/attributes acceptable by Cloud Pricing API. On Infracost official website, have a few exemples of parameters for specific properties, such as: _service, productFamily and attributes_.

![image](https://github.com/user-attachments/assets/14dc94fc-22ea-49d4-9b68-549cf1c13b8e)

Let me give an example of creating a pricing estimate to AWS EC2 instance. It resource (as well as many others) must be specified a lot of specific attributes to grant a good accuracy to the pricing estimate. However, I've been searching but so far I haven't found any documentation about all attributes that could be used and all parameters acceptable by each attribute.

Do you know if there is any way that could helps me on it?

**Comment by aliscott** [comment]
Hi @vitor-antoni, there's not an easy way to do this at the moment since so far we've only  concentrated on integrating it with the Infracost CLI, and not providing it as a standalone documented API.

The easiest way at the moment is to look at what the Infracost code does for the and build queries based on that, i.e. for AWS EC2 instances you can look in the `internal/resources/aws/instance.go` file and see [examples like this](https://github.com/infracost/infracost/blob/1f8f270f37dfc3900700c0ce445484f20f2eba04/internal/resources/aws/instance.go#L199-L219) which map directly to the pricing API lookups.

**Reply by vitor-antoni** [reply]
Okay, thank you!
